### PR TITLE
Add opacties data reader for h minus ions

### PR DESCRIPTION
**Pull request by light2802**
### :pencil: Description

**Type:**  :rocket: `feature` 

Add reader for parsing H minus opacities data files [1].

### :pushpin: Resources
[1] [Reference opacities data ](https://github.com/tardis-sn/carsus-data-negative-ions/blob/main/h_minus_cross_section_wbr.dat)

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by andrewfullard** [comment]
Tests would be great

File: carsus/io/opacities/opacities.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    h_minus_columns = ["wavelength", "cross_section"]
+
+    def __init__(self, fname=None):
+        """
+        Parameters
+        ----------
+        fname: str
+            Path to the h_minus file (http or local file).
+
+        priority: int, optional
```
Comment (jvshields) [inline carsus/io/opacities/opacities.py:36]: No priority argument passed in the declaration. Is this necessary for carsus? Either this should be added to the declaration or removed from the documentation.

File: carsus/io/opacities/opacities.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        Parameters
+        ----------
+        h_minus: pandas.DataFrame
+        selected_columns: list
+            list of which columns to select (optional - default=None which selects
+            a default set of columns)
+
+        Returns
+        -------
+            pandas.DataFrame
```
Comment (jvshields) [inline carsus/io/opacities/opacities.py:117]: Something is wrong here. Does this actually return a dataframe? A single column should be a series, and either way I didn't think pandas objects support astropy units so I don't think 129 works.

**Comment by jvshields** [comment]
I think that generally h_minus is not a good name for a data structure/reader, and also we should go back to the source of the h_minus_cross_section_wbr.dat file and carry that along here. I believe that this comes from bell_free-free_1980 in the stardis documentation, but if the reader is specific to this source then we should mention it at such. Alternatively, if this is a generic h minus ion reader (which is preferable), then many parts of the documentation need to be changed to not specify the h_minus_cross_section file by name (i.e., line 69 in opacities.py)

File: carsus/io/opacities/opacities.py (hunk 1 above)
Comment (chvogl) [reply carsus/io/opacities/opacities.py:36]: I don't think it is needed. I would just remove it from the docstring.

File: carsus/io/opacities/opacities.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+        Parameters
+        ----------
+        fname: ~str
+            path to h_minus_cross_section_wbr.dat
+
+        Returns
+        -------
+            pandas.DataFrame
+                pandas Dataframe represenation of h_minus
```
Comment (chvogl) [inline carsus/io/opacities/opacities.py:62]:
```suggestion
                pandas Dataframe representation of h_minus
```

File: carsus/io/opacities/opacities.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    """
+
+    h_minus_columns = ["wavelength", "cross_section"]
+
+    def __init__(self, fname=None):
+        """
+        Parameters
+        ----------
+        fname: str
```
Comment (chvogl) [inline carsus/io/opacities/opacities.py:33]:
```suggestion
        fname : str
```

File: carsus/io/opacities/opacities.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+logger = logging.getLogger(__name__)
+
+
+class HMINUSOPACITIESReader(object):
+    """
+    Class for extracting wavelengths and cross-section from h-minus opacities data files
+
+    Attributes
+    ----------
+    fname: path to h_minus_cross_section_wbr.dat
```
Comment (chvogl) [inline carsus/io/opacities/opacities.py:18]:
```suggestion
    fname : path to h_minus_cross_section_wbr.dat
```

File: carsus/io/opacities/opacities.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    """
+    Class for extracting wavelengths and cross-section from h-minus opacities data files
+
+    Attributes
+    ----------
+    fname : path to h_minus_cross_section_wbr.dat
+
+    Methods
+    --------
+    h_minus_raw:
```
Comment (jvshields) [inline carsus/io/opacities/opacities.py:22]: This is not a method of the class currently.

File: carsus/io/opacities/opacities.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        return self._h_minus
+
+    def read_h_minus(self, fname=None):
+        """
+        Reading in a normal h_minus_cross_section_wbr.dat
+
+        Parameters
+        ----------
+        fname : ~str
+            path to h_minus_cross_section_wbr.dat
```
Comment (jvshields) [inline carsus/io/opacities/opacities.py:57]: I still don't think this should be mentioned by name in the documentation. It should just be "path to h minus opacity file." 

Similarly for line 52 directly above.

**Comment by andrewfullard** [comment]
@jvshields is this going to be used for stardis? Does it need much more work?

**Pull request closed without merging**